### DISCUSSION TITLE
Accessing .Env when using —context argument

### DISCUSSION THREAD

**Discussion by assayire** [hairyhenderson/gomplate#1897] [Q&A] [answered]
Is it possible to access the `.Env` (say `.Env.USER`) when adding variables to root context?

For example:

```
# Assume name=Alice
gomplate \
  --context .=/var/tmp/context.yml \
  -i '{{ .name }} - {{ .Env.USER }}’
```

With the above setup, I am getting the following error:

```
Alice - 13:15:22 ERR  error="failed to render template <arg>: template: <arg>:1:24: executing \"<arg>\" at <.Env.USER>: map has no entry for key \"Env\""
```

**Comment by hairyhenderson** [comment, marked as answer]
No, because using `.` as the context name [overrides the root context](https://docs.gomplate.ca/usage/#--context-c).

However, instead of using `.Env` you can use [`env.Getenv`](https://docs.gomplate.ca/functions/env/#envgetenv) to (mostly) the same effect.

**Reply by assayire** [reply]
That’s awesome. Thanks.
